### DISCUSSION TITLE
submit pathway for consideration?  map of ADHD and Autism (ASD) markers

### DISCUSSION THREAD

**Discussion by alexmadsen1** [wikipathways/wikipathways-help#68] [Q&A] [answered]
I would greatly appreciate any guidance on how to submit this pathway for consideration and review. 

I am currently engaged in a project focused on mapping genetic pathways associated with ADHD (Attention-Deficit/Hyperactivity Disorder) and Autism Spectrum Disorder (ASD). Extensive studies suggest that both conditions have a strong genetic component, accounting for approximately 70% to 90% of their etiology.

My research has identified key genetic markers that are concentrated along specific metabolic and neurotransmitter pathways. These include:

One-Carbon (1C) Metabolism
-Folate Cycle
-Methionine Cycle
-Tetrahydrobiopterin (BH4) Pathway
Neurotransmitter Synthesis, Reuptake, and Breakdown
Neuroreceptor Dynamics in the Synaptic Cleft

My ultimate objective is to develop a comprehensive computational model that can be utilized in computational biology for further research and potential therapeutic interventions.

Attached is a screen capture of my current working draft: ASD & ADHD Related Pathways: One-carbon (1C); Folate, Methionine, BH4, Neurotransmitter: Synthesis, Reuptake, Breakdown, Neuroreceptors Synaptic Cleft
![ADHD ASD Pathway](https://github.com/wikipathways/wikipathways-help/assets/19489773/da14c6df-71b5-4c09-8f61-0b38d6d72ca9)

**Comment by alexmadsen1** [comment]
@khanspers, what do you suggest?

Alex

**Comment by khanspers** [comment]
Hi @alexmadsen1. Your pathway model looks great from the screenshot. In terms of content, the only comment I can make from seeing the screenshot is that perhaps the legend describing acronyms could be placed off to one side, or simply further down. 
Please go ahead and upload your pathway to WikiPathways (in PathVisio: Plugins > WikiPathways > Upload New). Some automated tests will then check for basic quality parameters like description, literature references, connected nodes etc. On a weekly basis, a human reviewer will also check your pathway and approve it if it meets our quality criteria:

- No unconnected interactions.
- Most (>90%) of datanodes must be annotated with database references.
- The pathway must have a complete description, meaning a 2-3 sentence summary of the process that the pathway describes.
- Minimum of one literature reference.
- Minimum of one pathway ontology tag.
- Overall, the pathway model should be useful for data visualization and analysis.

Note that you will need a WikiPathways account to upload the pathway: https://classic.wikipathways.org/index.php?title=Special:UserLogin&type=signup&returnto=WikiPathways

Kristina

**Comment by mkutmon** [comment]
The abbreviations could also be placed in the pathway description instead of in the diagram. Just a suggestion to clean up the visuals a bit.

**Comment by alexmadsen1** [comment]
Great! I will upload my working draft from my workstation this evening.  Yes, the abbreviations section is becoming inconveniently large.  I am breaking it up into smaller blocks.

**Comment by alexmadsen1** [comment]
When I try to upload my pathway file, I get the following error: 

Error while creating a new pathway as you do not have the write access. Please submit a request in the following link http://plugins.pathvisio.org/wp-client/request-webservice-access/ 

The link is dead: "This site can’t be reached."

Also, and perhaps related. I am getting an error when I try to Confirm e-mail address on https://classic.wikipathways.org/

Could not send confirmation mail. Check address for invalid characters.

Mailer returned: Failed to send data [SMTP: Invalid response code received from server (code: 554, response: Message rejected: Email address is not verified. The following identities failed the check in region US-WEST-2: noreply@wikipathways.org)]

**Reply by khanspers** [reply, marked as answer]
Hi @alexmadsen1, sorry about that. I have updated your account permissions (account: Alexmadsen1) so you should be able to upload now.

**Comment by khanspers** [comment]
Hi @alexmadsen1, thank you for submitting your [pathway model](https://classic.wikipathways.org/index.php/Pathway:WP5420). I have some feedback:
- There are a few unconnected interactions, for example from serotonin to the serotonin receptors. You can find all unconnected interactions by clicking Command+L (Mac) in PathVisio. 
- Many of the literature references are just URLs, which will be problematic on the new website. For example `https://imfar.confex.com/imfar/2008/webprogram/Paper2290.html` will not be recognized. Currently only PubMed, PMC, ISBN, and DOIs are recognized. Also, I believe DOIs would have to be in the form `DOI: 10.3390/nu8020068`, rather than just `10.3390/nu8020068` in order to be recognized on the new website. @egonw or @larsgw can confirm. 
- There are a few data nodes with fill color. Border color is a better choice since fill color will be lost if the pathway is used for typical data visualization, which often uses fill color. However, border color is tied to label color, so there are limitations to what will work in terms of readability. Either way, any special color should be defined in a legend in the pathway, and also mentioned in the description.
- Related, the colored interactions in the same area of the pathway (right side) are very hard to see. Switching to darker shades would work better.
- I also agree with Tina's comment above that the abbreviations should be part of the pathway description instead of being in the pathway.
- Finally, you noted the license as "Attribution-NonCommercial CC BY-NC". Please note that all WikiPathways content is distributed under [CC0](https://creativecommons.org/publicdomain/zero/1.0/). We encourage users to [cite individual pathways](https://www.wikipathways.org/cite.html), and quick access to full citations are available directly from each pathway page.

**Reply by alexmadsen1** [reply]
Thank you, this is very helpfull. I will get to work.

**Reply by khanspers** [reply]
To clarify about the first point, Command+L will highlight any unconnected interactions in green at the unconnected end.

**Comment by egonw** [comment]
> * Also, I believe DOIs would have to be in the form `DOI: 10.3390/nu8020068`, rather than just `10.3390/nu8020068` in order to be recognized on the new website. @egonw or @larsgw can confirm.

Actually, with the new website it has become easier, but even on the old website, it should be able to recognize the "10." as suggestion it is not a PubMed ID but a DOI. So, in the GPML it should look like this:

```xml
    <bp:PublicationXref xmlns:bp="http://www.biopax.org/release/biopax-level3.owl#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" rdf:id="ad2">
      <bp:ID rdf:datatype="http://www.w3.org/2001/XMLSchema#string">10.1211/PJ.2015.20068181</bp:ID>
      <bp:DB rdf:datatype="http://www.w3.org/2001/XMLSchema#string">DOI</bp:DB>
      <bp:TITLE rdf:datatype="http://www.w3.org/2001/XMLSchema#string">PCSK9 inhibitors: the next cholesterol-lowering blockbusters?</bp:TITLE>
      <bp:SOURCE rdf:datatype="http://www.w3.org/2001/XMLSchema#string">The Pharmaceutical Journal</bp:SOURCE>
      <bp:YEAR rdf:datatype="http://www.w3.org/2001/XMLSchema#string">2015</bp:YEAR>
      <bp:AUTHORS rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Dorey, Emma</bp:AUTHORS>
    </bp:PublicationXref>
```

See: https://github.com/wikipathways/wikipathways-database/blob/main/pathways/WP3408/WP3408.gpml#L197-L198

**Comment by alexmadsen1** [comment]
_[[khanspers](https://github.com/khanspers)]There are a few unconnected interactions, for example from serotonin to the serotonin receptors. You can find all unconnected interactions by clicking Command+L (Mac) in PathVisio.

[Alex] Done. I have used Ctrl L to find and fix all the unconnected interactions. If there are any remaining they are hiding from me. 

[[khanspers](https://github.com/khanspers)]Many of the literature references are just URLs, which will be problematic on the new website. For example https://imfar.confex.com/imfar/2008/webprogram/Paper2290.html will not be recognized. Currently only PubMed, PMC, ISBN, and DOIs are recognized. Also, I believe DOIs would have to be in the form DOI: 10.3390/nu8020068, rather than just 10.3390/nu8020068 in order to be recognized on the new website. @egonw or @larsgw can confirm.

[Alex] I have converted to the "DOI:" format where I found them as . The remaining web links do have a DOI, PubMed, PMC, or ISBN reference number that I could find.  My preference would be to keep we web link to maintain reference traceability unless it breaks something on the website badly. 

[[khanspers](https://github.com/khanspers)]There are a few data nodes with fill color. Border color is a better choice since fill color will be lost if the pathway is used for typical data visualization, which often uses fill color. However, border color is tied to label color, so there are limitations to what will work in terms of readability. Either way, any color should be defined in a legend in the pathway, and also mentioned in the description.
[Alex] Done. I have removed the special fill color.  Black now. 


[[khanspers](https://github.com/khanspers)]Related, the colored interactions in the same area of the pathway (right side) are very hard to see. Switching to darker shades would work better.
[Alex] I have not found a legible dark color scheme. Neurotransmitter Breakdown interaction is a rat's nest of overlapping interactions.  I have intentionally made the enzyme Catalysis lines light so they do not visually overwhelm the darker conversion interactions. If someone else knows has a recommendation on how to represent this Neurotransmitter Breakdown tangle of interactions cleanly while maintaining the detail and integrity of the interactions, I am open to suggestions. As this pathway mode is for computational analysis, network graph accuracy is the primary driver of diagram format, and visual readability is a secondary consideration. 

[[khanspers](https://github.com/khanspers)]I also agree with Tina's comment above that the abbreviations should be part of the pathway description instead of being in the pathway.

[Alex] Done.  I have moved the notes off the sheet. 

[[khanspers](https://github.com/khanspers)] Finally, you noted the license as "Attribution-NonCommercial CC BY-NC". Please note that all WikiPathways content is distributed under [CC0](https://creativecommons.org/publicdomain/zero/1.0/). We encourage users to [cite individual pathways](https://www.wikipathways.org/cite.html), and quick access to full citations is available directly from each pathway page._

[Alex] Done. I have changed it to CC0.

**Comment by khanspers** [comment]
Hi @alexmadsen1, thanks for making the updates. A couple of more things:
- The CC-BY license was still present in the Description from a Comment. I've removed it. 
- There is an unconnected line that appears to be going to an area where there might have been a node, that's now deleted. See attached.
- For a few of the nodes that are missing annotation, I was able to find identifiers. However, I noticed you were also editing at the same time so I didn't save my changes. I will update these later, unless you object: 
  - 4-aminobutanoate	
  - B-catenin -> CTNNB1
  - Cortactin -> CTTN
  - Fe2+
  - OH-
  - SLC18A2
  - SLC19A1
  - SLC6A2
![Screen Shot 2023-10-02 at 9 58 15 AM](https://github.com/wikipathways/wikipathways-help/assets/1442197/d2271d9f-4b9b-4f13-967f-3c2aaf03d309)

**Comment by alexmadsen1** [comment]
I am out of the pathway now. You are free to edit.

I have been setting up the Cofactors as complexes.  Originality I had them
as necessary stimulations, but I should think complexes are a better way to
represent cofactors.

Alex

**Comment by alexmadsen1** [comment]
@khanspers, what is the next step to get WP 5420 published?

**Comment by egonw** [comment]
hi @alexmadsen1, we have some automated testing you can have a look at: https://jenkins.bigcat.unimaas.nl/job/WikiPathways%20Curation%20(WP2RDF%20All%20Curation)/lastCompletedBuild/testReport/

You can find reports here like:

```
Ensembl identifiers for wrong species for a human pathway: 9. Details:
http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR6, ENSDARG00000016095
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR4, ENSDARG00000061940
 http://classic.wikipathways.org/instance/WP5420_r127488 -> ADRA1A, ENSDARG00000028725
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1D, ENSDARG00000054124
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR2a, ENSDARG00000057029
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR7, ENSDARG00000056949
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR2b, ENSDARG00000070940
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1E, ENSDARG00000011468
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1F, ENSDARG00000002773
```

```
Unexpected non-metabolite to metabolite conversions: 2. Details:
Homo sapiens http://classic.wikipathways.org/instance/WP5420_r127488 -> https://identifiers.org/uniprot/Q6ZNB7 https://identifiers.org/chebi/CHEBI:52392 http://rdf.wikipathways.org/Pathway/WP5420_r127488/WP/Interaction/idfbf6dbcd
```

```
Pathways DataNodes with WikiPathways ID that can be converted to have a RoundedRectangle ShapeType so that they become clickable: 2. Details:
http://classic.wikipathways.org/instance/WP5420_r127488 -> c487d, WP4271
 http://classic.wikipathways.org/instance/WP5420_r127488 -> e61d7, WP3869
  ==> expected: <0> but was: <2>
```

```
Unexpected metabolite to non-metabolite conversions:3 (See https://wikipathways.github.io/WikiPathwaysCurator/InteractionTests/noMetaboliteToNonMetaboliteConversions). . Details:
Homo sapiens http://classic.wikipathways.org/instance/WP5420_r127488 -> https://identifiers.org/wikidata/Q213050 https://identifiers.org/wikipathways/WP78 http://rdf.wikipathways.org/Pathway/WP5420_r127488/WP/Interaction/idae0a77d5
```

I can help. Some are more technical and easily fixed at GPML level, but harder with the PathVisio UI. If you are okay, I can fix some of them. For the long title, I leave that to the curation team how to handle that one.

**Reply by alexmadsen1** [reply]
@egonw  Egon,

You are welcome to fix the unexpected non-metabolite to metabolite
conversions.  I believe I fixed all the "Ensembl identifiers for wrong
species for a human pathway:" errors.

Alex

**Comment by danidi** [comment]
@alexmadsen1 you could remove "(Attention-Deficit/Hyperactivity Disorder)" from the title to make it shorter, as ADHD should be well known as abbreviation. If the full name is then mentioned in the description, the search should also work to find it with the full name. 

At the moment, the pathway states "No description", there are only comments in the description. As the pathway is too huge to easily grasp at first sight, I think it would be good to have a bit more of a description, e.g. on the aim of the pathway.

At the top right of the pathway is a node with pharmacological Intervention (drug), which is not connected to anything in the pathway. Is this on purpose? I guess it is meant as legend, but you could also mention this in the description instead, or at least convert it to a label.

What would be important to add is at least one pathway ontology tag in addition to the diseases ontology tags that you have already added. If you know the types of cells that are involved, you could also add a cell ontology term. 

A few more identifier issues I saw:

ADH protein has https://www.genome.jp/dbget-bin/www_bget?EC%201.1.1.1 as identifier (KEGG genes) but that doesn't resolve to anything. If that comes originally from another pathway, please let me know, we can then change it there as well. 

The metabolites AEA and metanephrine are added with two different identifiers (one from ChEBI, one from HMDB). While they are probably mapped in the background, it would still be good to stay consistent within a pathway.
FMN and FAD are also included with two different identifiers. In these cases, they point to different forms of the molecules (uncharged and 3-), which will look like they are two different molecules. Unless you want to differentiate the charges (then I would add the charge to the name as well), I would use a consistent identifier here. 

All "rs" RNA and gene products are included without database reference. @egonw do you know if we have a database that can be used for these?

**Comment by khanspers** [comment]
Excellent summary by @danidi ! 
For the rs numbers, I believe these are dbSNP ids, which is available in bridgedb. So they could all be assigned as "dbSNP".
Earlier [you commented](https://github.com/wikipathways/wikipathways-help/discussions/68#discussioncomment-7091556) on the multiple colored interaction lines in the upper right part, and I understand the point about keeping them light-colored to increase readability for the conversion lines. The different colors used seem to not have any specific meaning, other than which enzyme is responsible for the catalysis, if Im interpreting it right. It might be good to mention this somewhere, maybe in the description.

**Comment by alexmadsen1** [comment]
@egonw @danidi @khanspers Thank you for your review and feedback. 

 _For the rs numbers, I believe these are dbSNP ids, which is available in bridgedb. So they could all be assigned as "dbSNP"_.
 **[Alex] Great, I've been searching for a method to link the SNP to a database. Is there an automated approach or perhaps a table format I can use? I'd rather not manually map all 150 or so SNPs.**
- [DONE] You could remove "(Attention-Deficit/Hyperactivity Disorder)" from the title to make it shorter, as ADHD should be well known as abbreviation. If the full name is then mentioned in the description, the search should also work to find it with the full name.[DONE] At the moment, the pathway states "No description"; there are only comments in the description. As the pathway is too huge to easily grasp at first sight, I think it would be good to have a bit more of a description, e.g. on the aim of the pathway.
- [DONE, added more detail to legend and label] At the top right of the pathway is a node with pharmacological Intervention (drug), which is not connected to anything in the pathway. Is this on purpose? I guess it is meant as legend, but you could also mention this in the description instead, or at least convert it to a label.
- [done] At the moment, the pathway states "No description", there are only comments in the description. As the pathway is too huge to easily grasp at first sight, I think it would be good to have a bit more of a description, e.g. on the aim of the pathway.
- ADH protein has https://www.genome.jp/dbget-bin/www_bget?EC%201.1.1.1 as identifier (KEGG genes) but that doesn't resolve to anything. If that comes originally from another pathway, please let me know, we can then change it there as well.
- [Alex] EC 1.1.1.1 Enzyme · alcohol dehydrogenase; aldehyde reductase; ADH; alcohol dehydrogenase[Alex]  Unfortunately, this is the one area on the map where I haven't found a source to pinpoint a specific human gene or protein idea. All the research papers just reference the generic ALDH. EC 1.1.1.1 is the identifier for this class of enzime.
- [DONE], good catch] The metabolites AEA and metanephrine are added with two different identifiers (one from ChEBI, one from HMDB). While they are probably mapped in the background, it would still be good to stay consistent within a pathway.[DONE], came from different imports] FMN and FAD are also included with two different identifiers. In these cases, they point to different forms of the molecules (uncharged and 3-), which will look like they are two different molecules. Unless you want to differentiate the charges (then I would add the charge to the name as well), I would use a consistent identifier here.
- [Done] Earlier [you commented](https://github.com/wikipathways/wikipathways-help/discussions/68#discussioncomment-7091556) on the multiple colored interaction lines in the upper right part, and I understand the point about keeping them light-colored to increase readability for the conversion lines. The different colors used seem to not have any specific meaning, other than which enzyme is responsible for the catalysis, if Im interpreting it right. It might be good to mention this somewhere, maybe in the description.  
- [DONE] Ensembl identifiers for wrong species for a human pathway: 9. Details:
http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR6, ENSDARG00000016095
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR4, ENSDARG00000061940
 http://classic.wikipathways.org/instance/WP5420_r127488 -> ADRA1A, ENSDARG00000028725
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1D, ENSDARG00000054124
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR2a, ENSDARG00000057029
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR7, ENSDARG00000056949
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR2b, ENSDARG00000070940
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1E, ENSDARG00000011468
 http://classic.wikipathways.org/instance/WP5420_r127488 -> HTR1F, ENSDARG00000002773

**Comment by alexmadsen1** [comment]
Egon,

You are welcome to fix the unexpected non-metabolite to metabolite
conversions.  I believe if fixed all the "Ensembl identifiers for wrong
species for a human pathway:" errors.

Alex

**Comment by danidi** [comment]
Thank you for implementing the changes @alexmadsen1 
Here are a few more comments:
- The title now contains a typo (Autisum).
- I've found another issue with an identifier, cob(I)alamin has its name included as ChEBI ID
- the "pathway ontology" tag is very generic, would it be possible to specify a bit? e.g. at least a general term like "disease pathway", or maybe (if this fits) "metabolic pathway pertinent to the brain"?
- While adding the description, it seems the new lines in the list of the pathways were lost. It is now difficult to see where one pathway name ends and the next starts.
- For the enzyme class, I think it should be ok to use the EC number, but it seems this link is broken https://www.genome.jp/dbget-bin/www_bget?ec:1.1.1.1 works, but here https://www.genome.jp/dbget-bin/www_bget?EC%201.1.1.1 is used. @egonw is this a bridgeDB issue? Or is there a better datasource for EC numbers that can be used here?
- @egonw, do you have a possibility to automatically add the rs identifiers with dbSNP for all the rs names?

**Reply by alexmadsen1** [reply]
[Done] _The title now contains a typo (Autisum)._

 _the "pathway ontology" tag is very generic; would it be possible to specify a bit? e.g. at least a general term like "disease pathway", or maybe (if this fits) "metabolic pathway pertinent to the brain"?_
[Alex] This functionality on the website seems broken, search results do not return results for me, and the tree doest expand.

_While adding the description, it seems the new lines in the list of the pathways were lost. It is now difficult to see where one pathway name ends and the next starts._
[Alex] I have added commas as a separator.  The site keeps removing my carriage returns.

**Reply by danidi** [reply]
I can confirm the issue with the ontology functionality, I've opened a ticket here: https://github.com/wikipathways/wikipathways-help/issues/75

**Reply by alexmadsen1** [reply]
The ontology bug seems to have been resolved
[DONE] the "pathway ontology" tag is very generic; would it be possible to specify a bit? e.g. at least a general term like "disease pathway", or maybe (if this fits) "metabolic pathway pertinent to the brain"?

**Comment by alexmadsen1** [comment]
> * @egonw, do you have a possibility to automatically add the rs identifiers with dbSNP for all the rs names?

@danidi @egonw How do I modify the .GPML code outside the PathVisio WikiPathways plugin. Is there a recommended method to edit the .GPML without disrupting the WikiPathways plugin versioning? I'm considering writing a script to incorporate the rs identifiers from dbSNP."

**Reply by egonw** [reply]
hi @alexmadsen1, there are various approaches. We have been using scripts ourselves, and if you are comfortable with such approaches, you are most welcome to use them. Just be aware with more power comes more responsibility :)

Your options:

- the webservice, see https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006447 and https://webservice.wikipathways.org/ui/
- GitHub (but only for the pathways already there; tho this is where we are heading): https://github.com/wikipathways/wikipathways-database/blob/main/pathways/WP1000/WP1000.gpml
- Editing GPML on classic.wikipathways.org (tho for large pathways, you should copy/paste the GPML to a plain text editor), e.g. https://classic.wikipathways.org/index.php/Pathway:WP5420?action=edit

All have a bit of learning curve, but with scripting experience, it would not be too hard. The last option is what I would use for the DOI updates.

Maybe @AlexanderPico and @mkutmon have additional ideas?

**Reply by alexmadsen1** [reply]
@egonw I ended up linking the SNP to dbSNP by hand. Most of them are updated, but a few might have been missed as it was a manual process. 

@khanspers, to my knowledge, I have implemented all the requested corrections. What is the next step? 
Alex

**Reply by egonw** [reply]
@khanspers, let's take it forward.

**Reply by khanspers** [reply]
Hi @alexmadsen1, sorry for the late reply. I will switch the pathway to Approved on Monday morning (US pacific time). Thanks for the great work and the many edits!

**Reply by khanspers** [reply]
@alexmadsen1 : your pathway is now live on the new site https://www.wikipathways.org/pathways/WP5420.html. Some of the references are missing, because they are simple URLs (i.e. not DOI or PubMed). I believe this is being worked on and should be fixed soon.

**Reply by alexmadsen1** [reply]
Thank you for all your help and review. This is important, as shown by this week's news:

ADHD linked to increased dementia risk, study finds
"After researchers adjusted for 18 possible confounding factors, they found that adults with ADHD have a 2.77-fold increased risk of dementia compared to people without ADHD."
https://www.medicalnewstoday.com/articles/adhd-linked-to-increased-dementia-risk

**Comment by egonw** [comment]
> I believe this is being worked on and should be fixed soon.

Confirmed.

**Comment by samalita** [comment]
Hey @alexmadsen1 I came across this pathway and wonder if you can email me at samantha@adhdgirls.co.uk? I'm interested in more information and to see if you can expand this for women with ADHD and autism to include the hormonal pathways?
